### ISSUE TITLE
A failed statistical test leaves the previous result on screen with no indication it is stale

### ISSUE THREAD

**Issue by AshishK714** [social-engineer-ai/PyAnalytica#7]
**Labels:** bug, severity:medium, analyze, ux

## Environment

| | |
|---|---|
| PyAnalytica | 0.7.0 (installed from `dist/pyanalytica-0.7.0-py3-none-any.whl` into a clean venv) |
| Python | 3.14 |
| Key deps | shiny 1.7.0, pandas 3.0.5, numpy 2.5.2, scipy 1.18.1, matplotlib 3.11.1, seaborn 0.13.2 |
| OS / browser | Windows 11, Chrome |
| Dataset | `bank_campaign.csv` — 41,188 rows x 16 cols (UCI Bank Marketing, columns renamed) |

Found during a systematic 153-case sweep of every panel against one dataset.
Numbers were checked against an independent pandas/scipy/sklearn computation, not just
"did it render".

## Summary

When a test fails its input guard, the error is shown only as a ~5 second toast while the
results panel keeps rendering the **previous** test. Once the toast expires, the screen shows a
confident result that does not correspond to the inputs displayed beside it.

## Steps to reproduce

1. Load `bank_campaign.csv`, add a numeric column (or use `call_seconds`).
2. Analyze > Means > **Two-sample t-test**: Numeric = `subscribed_pct`, Group = `channel`.
   Run Test — succeeds.
3. Change Group Variable to `job` (12 levels). Run Test.

## Expected

The stale result is cleared, and the panel shows the error or an empty state.

## Actual

- A toast appears: *"Error: Expected 2 groups, got 12. Column 'job' has 12 unique values."*
  — this message is good and correctly identifies the problem.
- The results panel still shows the **previous** run: "The mean subscribed_pct for Landline
  (5.23) is significantly different from Mobile (14.74), t(41186) = -33.40 ...", with the
  sidebar reading Group Variable = `job`.
- After ~5 s the toast disappears and nothing on screen marks the result as stale.

## Cause

`mod_means.py::_run` catches the exception and calls `ui.notification_show(...)`, but never
resets `test_result_val`, so the `@render.ui` output keeps returning the last successful
result.

## Suggested fix

In the `except` branch, clear the stored result:

```python
except Exception as e:
    test_result_val.set(None)
    ui.notification_show(f"Error: {e}", type="error")
```

Better still, stamp each result with the inputs that produced it and show a "settings changed —
re-run" state when they diverge. The same pattern is worth auditing in the other Analyze and
Model modules.

## Evidence

`shots/D07_DEFECT_stale_result.jpg` — toast says "got 12" while the panel reports the
Landline-vs-Mobile result.

---

*Found in a systematic 153-case sweep of every panel against a 41,188-row teaching dataset, with each number checked against an independent pandas/scipy/scikit-learn computation rather than against the app itself.*
